### DISCUSSION TITLE
convert command is reducing the image size

### DISCUSSION THREAD

**Discussion by Prince7036** [ImageMagick/ImageMagick6#309] [General]
### ImageMagick version

ImageMagick 6.9.10-86

### Operating system, version and so on

linux

### Description

We are running the convert command to remove alpha channel

convert a.tif -alpha Off  output.tif

Older version of convert (ImageMagick 6.5.7-0) produces an output tif with size 873kb but when running the same command in newer version (ImageMagick 6.9.10-86) results in a output file with size 366KB

which is causing further issues in our process

Did anything change in the newer version ? If so how do we retain the older behaviour ?

**Comment by dlemstra** [comment]
This sounds like the newer version has better compression. Can you share the input image and both output images?

**Reply by Prince7036** [reply]
Unfortunately I can't share the actual image. But I can share the tiffinfo output if that would help

**Input tif file**
TIFF Directory at offset 0x6569a (415386)
  Subfile Type: (0 = 0x0)
  Image Width: 2550 Image Length: 3300
  Resolution: 300, 300 pixels/inch
  Bits/Sample: 8
  Compression Scheme: LZW
  Photometric Interpretation: RGB color
  Extra Samples: 1<unassoc-alpha>
  Samples/Pixel: 4
  Rows/Strip: 3300
  Planar Configuration: single image plane
  Predictor: horizontal differencing 2 (0x2)

**ImageMagick 6.9.10-86  output**
TIFF Directory at offset 0x5b558 (374104)
  Image Width: 2550 Image Length: 3300
  Resolution: 300, 300 pixels/inch
  Bits/Sample: 8
  Compression Scheme: LZW
  Photometric Interpretation: RGB color
  FillOrder: msb-to-lsb
  Orientation: row 0 top, col 0 lhs
  Samples/Pixel: 3
  Rows/Strip: 3300
  Planar Configuration: single image plane
  Page Number: 0-1
  White Point: 0.3127-0.329
  PrimaryChromaticities: 0.640000,0.330000,0.300000,0.600000,0.150000,0.060000
  Predictor: horizontal differencing 2 (0x2)

**ImageMagick 6.5.7-0**
TIFF Directory at offset 0xd38ca (866506)
  Image Width: 2550 Image Length: 3300
  Resolution: 300, 300 pixels/inch
  Bits/Sample: 8
  Compression Scheme: LZW
  Photometric Interpretation: RGB color
  FillOrder: msb-to-lsb
  Orientation: row 0 top, col 0 lhs
  Samples/Pixel: 3
  Rows/Strip: 1
  Planar Configuration: single image plane
  Page Number: 0-1
  DocumentName: test_ol6.tif
  Software: ImageMagick 6.5.7-0 2018-03-09 Q16 http://www.imagemagick.org
  Predictor: horizontal differencing 2 (0x2)

The major change I see is Rows/Strip which has not been changed in the newer version

**Comment by dlemstra** [comment]
As you pointed out Rows/Strip has changed between those version and without the files I cannot give you more information.

But I think it would be better to focus on the this comment instead
> which is causing further issues in our process

Can you explain in more detail what kind of issues are you having?

**Comment by fmw42** [comment]
The input file (top listing) has

```
Extra Samples: 1
Samples/Pixel: 4

```
6.9.10-86 has

```
Samples/Pixel: 3
Page Number: 0-1

```

6.5.7-0 has

```
Samples/Pixel: 3
Page Number: 0-1

```
So different versions show different number of channels or pages.  The top version has 4 channels and 1 extra channel. The bottom 2 have 3 channels and two pages.

I would not trust the bottom version of Imagemagick since it is so old. Lots of things have changed.

IM 6.9.10.86 is also a bit old.  You should upgrade if you can. We are currently at 6.9.13-7 or even better upgrade to IM 7.


> which is causing further issues in our process

What issues is this change causing you?  I would think a smaller file size would be desired.
